content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+
+In Meteor, persistent or shared data comes over the wire on publications. However, there are some types of data which doesn't need to be persistent or shared between users. For instance, the "logged-in-ness" of the current user, or the route they are currently viewing. 
+
+Although client-side state is often best contained as state of an individual template (and passed down the template heirarchy as arguments where necessary), sometimes you have a need for "global" state that is shared between unrelated sections of the template heirarchy.
+
+Usually such state is stored in a *global singleton* object which we can call a store. A singleton is a data structure of which only a single copy logically exists. The current user and the router from above are typical examples of such global singletons. 
+
+### Types of stores
+
+In Meteor, it's best to make stores *reactive data* sources, as that way they tie most naturally into the rest of the ecosystem. There are a few different packages you can use for stores.
+
+If the store is single-dimensional, you can probably use a `ReactiveVar` to store it (provided by the `reactive-var` package). A `ReactiveVar` has two properties, `get()` and `set()`:
+
+```js
+DocumentHidden = new ReactiveVar(document.hidden);
+$(window).on('visibilitychange', (event) => {
+  DocumentHidden.set(document.hidden);
+});
+```
+
+If the store is multi-dimensional, you may want to use a `ReactiveDict` (from the `reactive-dict` package):
+
+```js
+const $window = $(window);
+const getDimensions = () => {
+  return {
+    width: $window.width(),
+    height: $window.height()
+  };
+};
+
+WindowSize = new ReactiveDict(getDimensions());
+$window.on('resize', () => {
+  WindowSize.set(getDimensions());
+});
+```
+
+The advantage of a `ReactiveDict` is you can access each property individually (`WindowSize.get('width')`), and isolate the reactivity.
+
+If you need to query the store, or store many related items, it's probably a good idea to use a Local Collection (see the {% page_link collections "Collections Article" %}.
+
+### Accessing stores
+
+You should access stores in the same way you'd access other reactive data in your templates. For a Blaze template, that's either in a helper, or from within a `this.autorun()` inside an `onCreated()` callback. For React, it's from your `getMeteorData()` function. 
+
+This way you get the full reactive power of the store.
+
+### Updating stores
+
+If you need to update a store from as a result of user action, you'd update the store from an event handler, just like you call Methods. 
+
+If you need to perform complex logic in the update (i.e. not just call `.set()` etc), it's a good idea to define a mutator on the store. As the store is a singleton, you can just attach a function to the object directly:
+
+```js
+WindowSize.simulateMobile = (device) => {
+  if (device === 'iphone6s') {
+    this.set({width: 750, height: 1334});
+  }
+}
+```
+
+## Publishing Relational Data
```

Comment:
I feel like it's weird that this comes after client-side state... it's kind of a weird section in between a bunch of sections about server data and publications.